app/models/book_indexing.rb

```diff
@@ -0,0 +1,57 @@
+class BookIndexing
+  include Dynamoid::Document
+
+  VALID_STATES = [
+    PENDING = "pending",
+    STARTED = "started",
+    FINISHED = "finished"
+  ]
+  VALID_INDEXING_STRATEGIES = %w(I1)
+
+  validates :state, inclusion: { in: VALID_STATES }
+  validates :indexing_version, inclusion: { in: VALID_INDEXING_STRATEGIES }
+
+  validate :book_indexing_version_is_unique
+
+  field :state
+  field :book_version_id
+  field :indexing_version
+  field :enqueued_time, :datetime, store_as_string: true
+  field :started_time,  :datetime, store_as_string: true
+  field :finished_time, :datetime, store_as_string: true
+  field :message
+
+  # #create is used to add a book indexing dynamoid document to represent
+  # the enqueueing of a "job" to the SQS indexing pipeline
+  def self.create(book_version_id:, indexing_version:)
+    BookIndexing.new.tap do |job|
+      job.state = PENDING
+      job.book_version_id = book_version_id
+      job.indexing_version = indexing_version
+      job.enqueued_time = DateTime.now
+      job.save!
+    end
+  end
+
+  # #save is used to update a book indexing dynamoid document to represent
+  # the actual processing of a "job" in the SQS indexing pipeline
+  def start(book_version_id:)
+    # TODO
+  end
+
+  # #finish is used to update a book indexing dynamoid document to represent
+  # the completion a "job" in the SQS indexing pipeline
+  def finish(book_version_id:)
+    # TODO
+  end
+
+  private
+
+  def book_indexing_version_is_unique
+    found_books = BookIndexing.where(book_version_id: book_version_id,
+                                     indexing_version: indexing_version)
```

Comment:
I'm pretty sure this is not an efficient way to operate.  DynamoDB is a simple key value store and can't do efficient queries like a RDBMS.  Looking at the original Dynamoid blog post, a `where` query "does not efficiently perform a conditional join, but instead finds all the user's addresses and naively filters them in Ruby".  So combined with the loop below in `EnqueueIndexJobs` this means we are querying all of the book indexing records over and over.  I suspect we'll want to find a different approach.